### PR TITLE
Smart Budget

### PR DESCRIPTION
**Pull request by sahlminkok**
### Interactions
  - [ ] Splash screen
    - A simple page with the name of the app, and links to the signup and login pages.
  - [ ] Sign up and login pages
    - The user can register in the app with full name, email, and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
  - [ ] Home page (categories page)
    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
  - [ ] Transactions page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page, the user can see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
  - [ ] "Add a new category" page
    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
 - [ ] "Add a new transaction" page
   - The user fills out a form to create a new transaction with:
     - name (mandatory)
     - amount (mandatory)
     - categories (mandatory at least one)
   - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
   - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
### Testing requirements
  - [ ] I Create unit and integration tests for all the most important components of your RoR application.
### Technical Requirements
  - [ ] I use Postgres as my database.
  - [ ] I use devise for authentication.
  - [ ] I validate all user input to make sure that anyone with bad intentions cannot compromise my app.
  - [ ] I use a (.erb) view template engine
  - [ ] The project was deployed and accessible online.

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @sahlminkok 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- You ought to use GitFlow for this project instead of GitHub flow. The pull request you should submit should be from a feature branch intending to merge with the dev branch as the base branch. Kindly implement this and ensure that the feature branch contains all the commits that made this project. 👍 😎 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,24 +1,193 @@
...
+- **Expense Tracking**
+- **Categorized Transactions**
+- **Budget Insights**
+- **Mobile Accessibility**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://smartbudget-mhwp.onrender.com)
```
Comment (BregornOriginal) [inline README.md:73]: - [ ] Good work recording a presentation video, but there are some points missing. I would like to see how to create an account on your website at your presentation, and how it works. And please be simple when you mention a highlighting feature in your project. Remember to use less than 5 minutes in the video presentation too.

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @sahlminkok  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

one-step-closer-one-more

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by cynthiainga — APPROVED** [review]
Hi @sahlminkok, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] Add requested change ✔️ 
- [ ] No linters errors ✔️ 
- [ ] The app works as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**